indico/modules/events/abstracts/controllers/email_templates.py

```diff
@@ -0,0 +1,127 @@
+
+
+class RHEditEmailTemplateBase(RHManageAbstractsBase):
+    """Base class for operations that involve editing an e-mail template."""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.email_tpl
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageAbstractsBase._checkParams(self, params)
+        self.email_tpl = AbstractEmailTemplate.get_one(request.view_args['email_tpl_id'])
+
+
+class RHEditEmailTemplateRules(RHEditEmailTemplateBase):
+    """Edit the rules of a notification template."""
+
+    def _process(self):
+        form = EditEmailTemplateRuleForm(obj=self.email_tpl, event=self.event_new)
+        if form.validate_on_submit():
+            form.populate_obj(self.email_tpl)
+            tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+            return jsonify_data(html=tpl.render_notification_list(self.event_new))
+        return jsonify_template('events/abstracts/management/notification_tpl_form.html',
+                                form=form, is_edit=True)
+
+
+class RHEditEmailTemplateText(RHEditEmailTemplateBase):
+    """Edit the e-mail text of a notification template."""
+
+    def _process(self):
+        form = EditEmailTemplateTextForm(obj=self.email_tpl, event=self.event_new)
+        if form.validate_on_submit():
+            form.populate_obj(self.email_tpl)
+            tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+            return jsonify_data(html=tpl.render_notification_list(self.event_new))
+        return jsonify_template('events/abstracts/management/notification_tpl_text_form.html',
+                                form=form, is_edit=True)
+
+
+class RHSortEmailTemplates(RHManageAbstractsBase):
+    """Sort e-mail templates according to the order provided by the client."""
+
+    def _process(self):
+        new_order = [AbstractEmailTemplate.get(tpl_id) for tpl_id in request.json['sort_order']]
+        if set(new_order) != set(self.event_new.abstract_email_templates):
+            raise ValueError('Sorted and stored lists have different contents.')
+        for position, tpl in enumerate(new_order):
+            tpl.position = position
+        return jsonify_data()
+
+
+class RHDeleteEmailTemplate(RHEditEmailTemplateBase):
+    """Delete an e-mail template."""
+
+    def _process(self):
+        db.session.delete(self.email_tpl)
+        tpl = get_template_module('events/abstracts/management/_notification_tpl_list.html')
+        return jsonify_data(html=tpl.render_notification_list(self.event_new),
+                            flash=False)
+
+
+class RHPreviewEmailTemplate(RHEditEmailTemplateBase):
+    """Preview an e-mail template."""
+
+    def _process(self):
+        abstract = create_mock_abstract(self.event_new)
+        body = replace_placeholders('abstract-notification-email', self.email_tpl.body,
```

Comment:
Yep, it would be nice. But I think I'll leave that for a 2nd round. This PR is getting way too big and GitHub is already having issues rendering it 😄